go junitxml: compute per package start times rather than using per execution

zdylag opened the pull request:
This can be pretty meaningful if people want to build a timeline of when what test packages were executing. One example use case would be building a trace view ala https://docs.datadoghq.com/tests/setup/junit_xml/?tab=linux.

dnephin commented on testjson/execution.go:
```diff
@@ -131,6 +131,23 @@ func (p *Package) TestCases() []TestCase {
 
+// EarliestTime returns the earliest start time found within a child test case,
+// and a bool for if any such time was found.
+func (p *Package) EarliestTime() (time.Time, bool) {
+	found := false
+	t := time.Time{}
```
A more descriptive name for `t` would be great. Maybe `earliest` ?

dnephin commented on testjson/execution.go:
```diff
@@ -131,6 +131,23 @@ func (p *Package) TestCases() []TestCase {
+		found = true
+	}
+
+	return t, found
```
I think we don't need the bool return value here. Checking if the returned value `IsZero` should be fine.

zdylag replied:
I actually originally did that, but thought it was more likely to get flagged in the PR 😅 Sounds good I will change both of these 👍

dnephin reviewed (commented):
Thank you for the PR! This fix sounds good

zdylag commented:
Hey @dnephin ! Sorry for the super slow turnaround, but I finally addressed your feedback 😅 No rush clearly, but just letting you know it's been updated.

dnephin commented:
Thank you for the PR! While reviewing this again I realized there are other events that wouldn't be captured in `TestCases` that could include an earlier timestamp.   If I run `go test -json`, I see an `Action=Start` event for packages.


I kept your commit, and I opened #482 to use any event time for both the start and end time. This way time reported by the CLI will match the time reported by junitxml.

If that works for you I can tag a release with the change.
